sdk/formrecognizer/ai-form-recognizer/src/documentAnalysisClient.ts

```diff
@@ -0,0 +1,594 @@
+   * console.log("The type of this receipt is:", receipt.receiptType?.value);
+   * ```
+   *
+   * #### From Request Body
+   *
+   * Alternatively, if the file is local (or in memory in the browser), a binary object can be uploaded. The following
+   * example uses the Node.JS filesystem API.
+   *
+   * ```javascript
+   * import * as fs from "fs";
+   * import { PrebuiltModels } from "@azure/ai-form-recognizer";
+   *
+   * const file = fs.createReadStream("path/to/receipt.pdf");
+   *
+   * // The model that is passed to the following function call determines the type of the eventual result. In the
+   * // example, we will use the prebuilt receipt model.
+   * const poller = await client.beginAnalyzeDocuments(PrebuiltModels.Receipt, file);
+   *
+   * // The result is a long-running operation (poller), which must itself be polled until the operation completes
+   * const {
+   *   pages, // pages extracted from the document, which contain lines and words
+   *   tables, // extracted tables, organized into cells that contain their contents
+   *   styles, // text styles (ex. handwriting) that were observed in the document
+   *   keyValuePairs, // extracted pairs of elements  (directed associations from one element in the input to another)
+   *   entities, // extracted entities in the input's content, which are categorized (ex. "Location" or "Organization")
+   *
+   *   documents // extracted documents (instances of one of the model's document types and its field schema)
+   * } = await poller.pollUntilDone();
+   *
+   * // Extract the fields of the first document. These fields constitute a receipt, because we used the receipt model
+   * const [{ fields: receipt }] = documents;
+   *
+   * // Since we used the strongly-typed PrebuiltModels.Receipt object instead of the "prebuilt-receipt" model ID
+   * // string, the fields of the receipt are strongly-typed and have camelCase names (as opposed to PascalCase).
+   * console.log("The type of this receipt is:", receipt.receiptType?.value);
+   * ```
+   *
+   * @param model - a {@link DocumentModel} representing the model to use for analysis and the expected output type
+   * @param input - a URL (string) to an input document accessible from the public internet, or a
+   *                {@link FormRecognizerRequestBody} that will be uploaded with the request
+   * @param options - optional settings for the analysis operation and poller
+   * @returns a long-running operation (poller) that will eventually produce an `AnalyzeResult` with documents that have
+   *          the result type associated with the input model
+   */
+  public async beginAnalyzeDocuments<Document>(
+    model: DocumentModel<Document>,
+    input: string | FormRecognizerRequestBody,
+    // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
+    options?: AnalyzeDocumentsOptions<AnalyzeResult<Document>>
```

Comment:
are there any options for the polling path the could be added to the options bag here? same question for all LRO operations.

Reply:
Could you clarify what you mean by polling path?

Reply:
That is the old swagger. In any case, those parameters are destructured from the operation-location. The only other options are just OperationOptions, so I think we're covered here.

Reply:
Sounds good. I would keep an eye on any future option to make sure the SDK supports it.